### PR TITLE
Additional rules search

### PR DESCRIPTION
**Pull request by dogversioning**
This PR makes the following changes:
-Adds tables for discovering rules from additional sources
-Temporarily adds an RXNConso table locally, which will be removed in a followon commit
-Includes some midflight changes for the local mysql version being used for the current paper

### CODE REVIEW

File: cumulus_library_opioid/vocab/analyze/curate.sql
Comment (dogversioning) [inline cumulus_library_opioid/vocab/analyze/curate.sql:1]: Noting this as a start point for the midflight changes

File: cumulus_library_opioid/vocab/make_all.sh
Comment (dogversioning) [inline cumulus_library_opioid/vocab/make_all.sh:1]: noting this as the end for midflight changes

File: cumulus_library_opioid/vocab/rxnorm_vsac_builder.py [hunk 1]
```diff
@@ -19,65 +19,90 @@ def prepare_queries(
...
+                    column_aliases = {"b.str": "keyword"}
                 )
             )
+            # TODO: Fix this template to mimic the static rxnconso file
+            # self.queries.append(
+            #     base_templates.get_base_template(
+            #         'create_annotated_rxnconso',
+            #         self.base_path / "template_sql",
+            #         steward=steward
+            #     )
```
Comment (dogversioning) [inline cumulus_library_opioid/vocab/rxnorm_vsac_builder.py:77]: This has been fussy enough that I wanted to get this in and then come back and do the next step as a seperate effort - you could talk me into trying to do it now, but that was my thought.

File: cumulus_library_opioid/vocab/vsac.py [hunk 2]
```diff
@@ -63,10 +63,9 @@ def download_oid_data(
 
     response = api.get_vsac_valuesets(oid=VSAC_OIDS[source_vocab])
     for valueset in response:
-        include = valueset.get("compose").get("include", [])
-        for data in include:
-            for concept in data.get("concept", []):
-                output.append([concept["code"], concept["display"]])
+        contains = valueset.get("expansion").get("contains", [])
+        for data in contains:
+            output.append([data["code"], data["display"]])
```
Comment (dogversioning) [inline cumulus_library_opioid/vocab/vsac.py:68]: this is due to the related change to VSAC in the core library

File: references/BuprenorphineQuickStartGuideIndicationsMatrixCOWS.pdf
Comment (dogversioning) [inline references/BuprenorphineQuickStartGuideIndicationsMatrixCOWS.pdf:1]: These three PDFs were commited twic, the 'citations' location is andy's preferred location.

File: tests/test_data/rxnrel.csv [hunk 3]
```diff
@@ -1,12 +1,12 @@
 "rxcui1","rxaui1","stype1","rel","rxcui2","rxaui2","stype2","rela","rui","srui","sab","sl","rg"
-"1819",,"CUI","RN","1818",,"CUI","tradename_of","4716626",,"RXNORM",,
-"1819",,"CUI","RN","203840",,"CUI","tradename_of","4317610",,"RXNORM",,
+"1819",,"CUI","RN","1818",,"CUI","reformulated_to","4716626",,"RXNORM",,
+"1819",,"CUI","RN","203840",,"CUI","reformulated_to","4317610",,"RXNORM",,
```
Comment (dogversioning) [inline tests/test_data/rxnrel.csv:3]: This change was to make the included_rels table actually detect something.

File: cumulus_library_opioid/reference_sql/static_builder.sql [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+-- and it may not be correct for all databases. Use the CLI's build 
+-- option to derive the best SQL for your dataset.
+
+-- ###########################################################
+
+CREATE EXTERNAL TABLE IF NOT EXISTS `cumulus_mhg_dev_db`.`opioid__keywords` (
+    STR STRING
+)
+STORED AS PARQUET
+LOCATION 's3://cumulus-athena-933137588087-us-east-1/results/cumulus_user_uploads/cumulus_mhg_dev_db/opioid/keywords.filtered'
```
Comment (mikix) [inline cumulus_library_opioid/reference_sql/static_builder.sql:13]: Do we want URLs like this in the public repo?

File: cumulus_library_opioid/vocab/analyze/curate.sql
Comment (mikix) [reply cumulus_library_opioid/vocab/analyze/curate.sql:1]: > some midflight changes for the local mysql version being used for the current paper

That sounds like these changes are to deal with a new database backend's (mysql) SQL quirks. But these look like logic and keyword changes. I mean it seems fine, I'm just not clear on what is midflight / what the end goal is.

File: cumulus_library_opioid/vocab/rxnorm_vsac_builder.py [hunk 5]
```diff
@@ -19,65 +19,90 @@ def prepare_queries(
...
-                b_table= b_table or f'opioid__{steward}_vsac',
-                a_join_col= a_join_col or 'rxcui',
-                b_join_col = b_join_col or 'code',
-                **kwargs
+                tables = [
+                    f'{a_schema}{a_table}',
+                    b_table or f'opioid__{steward}_vsac',
+                ],
+                table_aliases = ['a','b'],
+                join_clauses=join_clauses or [f"{a_join_col or 'a.rxcui'} = {b_join_col or 'b.code'}"],
```
Comment (mikix) [inline cumulus_library_opioid/vocab/rxnorm_vsac_builder.py:45]: This line feels a bit awkward to me. Maybe move the arg-defaulting logic up to the top of the method? Like:

```
a_join_col = a_join_col or 'a.rxcui'
b_join_col = b_join_col or 'b.code'
join_clauses = join_clauses or [f"{a_join_col} = {b_join_col}"]
... 
get_create_view_from_tables(
  ...
  join_clauses=join_clauses,
)
```

Which feels.... slightly better

File: cumulus_library_opioid/vocab/static_builder.py [hunk 6]
```diff
@@ -30,27 +30,34 @@ class StaticBuilder(base_table_builder.BaseTableBuilder):
...
+            table_name="search_rules",
             headers=[
                 "TTY1",
                 "RELA",
                 "TTY2",
                 "rule",
             ],
             dtypes={"TTY1": "str", "RELA": "str", "TTY2": "str", "rule": "str"},
-            parquet_types=["STRING", "STRING", "STRING", "STRING", "STRING"],
+            parquet_types=["STRING", "STRING", "STRING", "STRING", "BOOLEAN"],
```
Comment (mikix) [inline cumulus_library_opioid/vocab/static_builder.py:60]: What's happening here. Looks like there are four columns, all strings. But then there's a fifth bool column now?

File: cumulus_library_opioid/vocab/static_builder.py [hunk 7]
```diff
@@ -176,4 +194,4 @@ def prepare_queries(
                         remote_table_cols_types=table.parquet_types,
                     )
                 )
-                progress.advance(task)
+                progress.advance(task)
```
Comment (mikix) [inline cumulus_library_opioid/vocab/static_builder.py:197]: :scissors:

File: cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    r.rui,
+    r.rel,
+    r.rela,
+    r.str1,
+    r.str2,
+    r.keyword
+FROM opioid__{{ steward }}_potential_rules AS r
+WHERE
+    r.REL NOT IN ('RB', 'PAR')
+    AND length(r.keyword) >= 4
```
Comment (mikix) [inline cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja:18]: Sorry, this struck me as odd here - we control the keywords, yeah? Why not just not add small ones to the list?

File: tests/test_data/rxnrel.csv (hunk 3 above)
Comment (mikix) [reply tests/test_data/rxnrel.csv:3]: Was this file generated/downloaded originally? Maybe add the fact that you made this edit to the generation instructions too.

File: tests/test_vsac.py [hunk 9]
```diff
@@ -34,10 +34,11 @@ def test_download_oid_data(mock_api, mock_db, name, umls, force, raises, tmp_pat
         )
         vsac.download_oid_data(name,path=tmp_path, config=config)
         output_dir = list(tmp_path.glob("*"))
-        assert len(output_dir) == 5
+        print(output_dir)
```
Comment (mikix) [inline tests/test_vsac.py:37]: :scissors:

File: tests/test_vsac.py [hunk 10]
```diff
@@ -34,10 +34,11 @@ def test_download_oid_data(mock_api, mock_db, name, umls, force, raises, tmp_pat
...
+        assert len(output_dir) == 4
```
Comment (mikix) [inline tests/test_vsac.py:38]: Why not just check the actual filenames? Those feel like we'd want to notice a change there too, not just in the number of files.

**Review by mikix — APPROVED** [review]

File: cumulus_library_opioid/reference_sql/static_builder.sql (hunk 4 above)
Comment (dogversioning) [reply cumulus_library_opioid/reference_sql/static_builder.sql:13]: probably not - i'll  hand wave it for now, but i'll have to change the behavior in the library so this isn't re-introduced

File: cumulus_library_opioid/vocab/analyze/curate.sql
Comment (dogversioning) [reply cumulus_library_opioid/vocab/analyze/curate.sql:1]: there may be additional changes coming - andy is still actively working on this, but now on another branch. The queries being created in the templates should line up with these.

File: cumulus_library_opioid/vocab/rxnorm_vsac_builder.py (hunk 5 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/rxnorm_vsac_builder.py:45]: can do.

File: cumulus_library_opioid/vocab/static_builder.py (hunk 6 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/static_builder.py:60]: the source file has four columns. the map_cols structure below allows you to pivot one column to another, and map over a subset of values to the new column, generating that extra column.

File: cumulus_library_opioid/vocab/static_builder.py (hunk 7 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/static_builder.py:197]: what a good reminder to do a ruff pass, got too used to it being automated.

File: cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja (hunk 8 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja:18]: Right now, there are none that are smaller than 4 chars. I'm not sure if the intent is to force keywords to be relatively large? or if it's just 'mapping this logic to the data we have? @comorbidity

File: tests/test_data/rxnrel.csv (hunk 3 above)
Comment (dogversioning) [reply tests/test_data/rxnrel.csv:3]: ok, i'll add a note someplace

File: tests/test_vsac.py (hunk 10 above)
Comment (dogversioning) [reply tests/test_vsac.py:38]: sure, can do.

File: cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja (hunk 8 above)
Comment (dogversioning) [reply cumulus_library_opioid/vocab/template_sql/create_included_keywords.sql.jinja:20]: per offline discussion, added a comment about this for future reference

**Pull request merged**